### ISSUE TITLE
Autodoc is broken

### ISSUE THREAD

**Issue by HugoStrand** [TRIQS/dft_tools#100]
Dear all,

The sphinx auto doc of the python code is not working, making the reference documentation rather useless, see: 

https://triqs.github.io/dft_tools/master/reference/sumk_dft.html
https://triqs.github.io/dft_tools/master/reference/sumk_dft_tools.html
https://triqs.github.io/dft_tools/master/reference/transbasis.html
https://triqs.github.io/dft_tools/master/reference/block_structure.html

Maybe time to fix?

Best, Hugo

**Comment by mzingl** [comment]
First, congratulation to opening the 100th issue! :1st_place_medal:
I have currently problems to in compile the TRIQS doc on my machine, but as soon as this works, I 'll have a look.

Best,
Manuel

**Comment by Wentzell** [comment]
I can reproduce this problem on my local machine when you building in an environment without an installed version of dft_tools.

It is linked to the fact that autodoc needs to have a working version of dft_tools available for all the imports.
Cf. https://jenkins.flatironinstitute.org/job/TRIQS/job/dft_tools/job/unstable/140/execution/node/64/log/

```
/src/dft_tools/doc/reference/sumk_dft.rst:5: WARNING: autodoc: failed to import class u'SumkDFT' from module u'triqs_dft_tools.sumk_dft'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 386, in import_object
    __import__(self.modname)
ImportError: No module named triqs_dft_tools.sumk_dft
/src/dft_tools/doc/reference/sumk_dft_tools.rst:5: WARNING: autodoc: failed to import class u'SumkDFTTools' from module u'triqs_dft_tools.sumk_dft_tools'; the following exception was raised:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/ext/autodoc.py", line 386, in import_object
    __import__(self.modname)
ImportError: No module named triqs_dft_tools.sumk_dft_tools
```

The proper solution will be to adjust the project structure such that you construct a working version of dft_tools (i.e. the modules) during build-time in the CMAKE_BINARY_DIR and import from there during the Doc creation.

**Comment by gkraberger** [comment]
Hi Nils,

this fix is included in my PR #105.

Please merge!
